tests/test_update_runnablejobs.py

```diff
@@ -0,0 +1,175 @@
+    """
+
+    results = []
+    for platform_option in ["debug", "opt", "asan"]:
+        for build_system_type in ["buildbot", "taskcluster"]:
+            if build_system_type == "buildbot":
+                for build_platform in ["windows8-64", "windowsxp"]:
+                    # It's little different from the really ref_data_name and buildername
+                    # for buildbot, but I think it's enough for a test case.
+                    temp = "{build_platform} {platform_option} test test-for-{build_platform}"
+                    ref_data_name = temp.format(build_platform=build_platform,
+                                                platform_option=platform_option)
+                    results.append({'build_system_type': build_system_type,
+                                    'platform_option': platform_option,
+                                    'ref_data_name': ref_data_name,
+                                    'build_platform': build_platform})
+            else:
+                for build_platform in ["linux64", "android-4-3-armv7-api15"]:
+                    temp = "desktop-test-{build_platform}/" \
+                           "{platform_option}-test-for-{build_platform}"
+                    ref_data_name = temp.format(build_platform=build_platform,
+                                                platform_option=platform_option)
+                    results.append({'build_system_type': build_system_type,
+                                    'platform_option': platform_option,
+                                    'job_type_name': ref_data_name,
+                                    'ref_data_name': ref_data_name,
+                                    'build_platform': build_platform})
+
+    return {"results": results}
+
+
+class TestUpdateRunnableapi(unittest.TestCase):
+    """This class is used to test the update_runnableapi function"""
+
+    MOCK_DATA = data_generator()
+    MOCK_PUSH = {"taskId": 123456, "expires": "2017-07-04T22:13:23"}
+
+    def setUp(self):
+        """Setting up values that will be used in every test."""
+
+        self.expected = self.MOCK_DATA
+
+        # we need to create two timetamp to mock the new one and old one
+        today = datetime.date.today().strftime('%Y-%m-%d')
+        time_tuple = datetime.datetime.strptime(today, "%Y-%m-%d").timetuple()
+        old_time = (datetime.date.today() -
+                    datetime.timedelta(days=1)).strftime('%Y-%m-%d')
+        old_time_tuple = datetime.datetime.strptime(old_time, "%Y-%m-%d").timetuple()
+        self.new_timestamp = time.mktime(time_tuple)
+        self.old_timestamp = time.mktime(old_time_tuple)
+
+        # rename the runnablejob.json and preseed.json file untill the end of test
+        try:
+            if os.path.isfile(ORI_RUNNABLE_PATH):
+                os.rename(ORI_RUNNABLE_PATH, TMP_RUNNABLE_FILENAME)
+            shutil.copy2(ORI_PRESEED_PATH, TMP_PRESEED_FILENAME)
```

Comment:
yep, we suppose the runnablejobs.json wasn't exist in the first place(and we make sure that before we begin the test https://github.com/mozilla/ouija/pull/110/files#diff-7cd01f67f0e148d273f72b8676ee7907R134). But we should keep the preseed.json we had :)